org.osgi.test.cases.cm/src/org/osgi/test/cases/cm/junit/CMControl.java

```diff
@@ -2294,10 +2367,10 @@ public void updated(final Dictionary<String, ?> properties) throws Configuration
 			assertNull(updates.get(0));
 			assertEquals("somevalue", updates.get(1).get("scalar"));
 			assertEquals("newvalue", updates.get(2).get("scalar"));
-		} finally {
-			this.unregisterService(listener);
-			this.unregisterService(ms);
-		}
+			reg.unregister();
+			// TODO BJ:WHY MUST THIS BE DONE MANUALLY????
```

Comment:
I assume that since the test method uses the raw service API to register a service, it must then unregister it as no one else will.